engine/access/ingestion2/engine.go

```diff
@@ -0,0 +1,180 @@
+	receipts                 storage.ExecutionReceipts
+	collectionExecutedMetric module.CollectionExecutedMetric
+}
+
+var _ network.MessageProcessor = (*Engine)(nil)
+
+func New(
+	log zerolog.Logger,
+	net network.EngineRegistry,
+	finalizedBlockProcessor *FinalizedBlockProcessor,
+	collectionSyncer *CollectionSyncer,
+	receipts storage.ExecutionReceipts,
+	collectionExecutedMetric module.CollectionExecutedMetric,
+) (*Engine, error) {
+	executionReceiptsRawQueue, err := fifoqueue.NewFifoQueue(defaultQueueCapacity)
+	if err != nil {
+		return nil, fmt.Errorf("could not create execution receipts queue: %w", err)
+	}
+	executionReceiptsQueue := &engine.FifoMessageStore{FifoQueue: executionReceiptsRawQueue}
+	messageHandler := engine.NewMessageHandler(
+		log,
+		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*flow.ExecutionReceipt)
+				return ok
+			},
+			Store: executionReceiptsQueue,
+		},
+	)
+
+	e := &Engine{
+		log:                      log.With().Str("engine", "ingestion2").Logger(),
+		finalizedBlockProcessor:  finalizedBlockProcessor,
+		collectionSyncer:         collectionSyncer,
+		messageHandler:           messageHandler,
+		executionReceiptsQueue:   executionReceiptsQueue,
+		receipts:                 receipts,
+		collectionExecutedMetric: collectionExecutedMetric,
+	}
+
+	// register our workers which are basically consumers of different kinds of data.
+	// engine notifies workers when new data is available so that they can start processing them.
+	builder := component.NewComponentManagerBuilder().
+		AddWorker(e.startMessageHandlerLoop).
+		AddWorker(e.finalizedBlockProcessor.StartWorkerLoop).
+		AddWorker(e.collectionSyncer.StartWorkerLoop)
+	e.ComponentManager = builder.Build()
+
+	// engine gets execution receipts from channels.ReceiveReceipts channel
+	_, err = net.Register(channels.ReceiveReceipts, e)
+	if err != nil {
+		return nil, fmt.Errorf("could not register engine in network to receive execution receipts: %w", err)
+	}
+
+	return e, nil
+}
+
+// Process processes the given event from the node with the given origin ID in
+// a blocking manner. It returns the potential processing error when done.
+//
+// No errors are expected during normal operations.
+func (e *Engine) Process(chanName channels.Channel, originID flow.Identifier, event interface{}) error {
+	select {
+	case <-e.ComponentManager.ShutdownSignal():
+		return component.ErrComponentShutdown
+	default:
+	}
+
+	switch event.(type) {
+	case *flow.ExecutionReceipt:
+		err := e.messageHandler.Process(originID, event)
+		return err
+	default:
+		return fmt.Errorf("got invalid event type (%T) from %s channel", event, chanName)
+	}
+}
+
+// startMessageHandlerLoop reacts to message handler notifications and processes available execution receipts
+// once notification has arrived.
+func (e *Engine) startMessageHandlerLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
```

Comment:
I think using start in the name is confusing. the component's `Start` method is non-blocking and kicks off the worker within a gorountine. This method is blocking. We typically use naming like this in the code base.
```suggestion
func (e *Engine) messageHandlerLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
```

Reply:
sure. will change it